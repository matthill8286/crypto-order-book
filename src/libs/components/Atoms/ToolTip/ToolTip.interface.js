
//# sourceMappingURL=ToolTip.interface.js.map