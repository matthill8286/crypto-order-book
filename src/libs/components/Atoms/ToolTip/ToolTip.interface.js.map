{"version":3,"sources":[],"names":[],"mappings":"","sourcesContent":[],"file":"ToolTip.interface.js"}