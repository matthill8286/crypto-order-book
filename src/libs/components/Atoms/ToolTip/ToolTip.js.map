{"version":3,"sources":["../../../../src/components/Atoms/ToolTip/ToolTip.tsx"],"names":["React","useState","InfoText","StyledToolTip","StyledToolTipContent","StyledToolTipWrapper","ToolTip","props","children","content","primary","limitLines","otherProps","tooltipVisible","setState"],"mappings":";;;;;;;;;;;;;;;;;;AAAA,OAAOA,KAAP,IAAgBC,QAAhB,QAAgC,OAAhC;AACA,SAASC,QAAT;AAEA,SAASC,aAAT,EAAwBC,oBAAxB,EAA8CC,oBAA9C;AAEA,OAAO,IAAMC,OAA+B,GAAG,SAAlCA,OAAkC,CAAAC,KAAK,EAAI;AAAA,MAC9CC,QAD8C,GACwBD,KADxB,CAC9CC,QAD8C;AAAA,MACpCC,OADoC,GACwBF,KADxB,CACpCE,OADoC;AAAA,uBACwBF,KADxB,CAC3BG,OAD2B;AAAA,MAC3BA,OAD2B,+BACjB,KADiB;AAAA,0BACwBH,KADxB,CACVI,UADU;AAAA,MACVA,UADU,kCACG,CADH;AAAA,MACSC,UADT,4BACwBL,KADxB;;AAAA,kBAEnBN,QAAQ,CAAC,KAAD,CAFW;AAAA;AAAA,MAE/CY,cAF+C;AAAA,MAE/BC,QAF+B;;AAItD,SACE,oBAAC,oBAAD;AACE,IAAA,YAAY,EAAE;AAAA,aAAMA,QAAQ,CAAC,IAAD,CAAd;AAAA,KADhB;AAEE,IAAA,YAAY,EAAE;AAAA,aAAMA,QAAQ,CAAC,KAAD,CAAd;AAAA,KAFhB;AAGE,IAAA,OAAO,EAAE;AAAA,aAAMA,QAAQ,CAAC,CAACD,cAAF,CAAd;AAAA;AAHX,KAIMD,UAJN,GAKGJ,QALH,EAMGK,cAAc,IAAIJ,OAAlB,IACC,oBAAC,aAAD;AAAe,IAAA,OAAO,EAAEC;AAAxB,KACE,oBAAC,oBAAD;AAAsB,IAAA,OAAO,EAAEA;AAA/B,KACE,oBAAC,QAAD;AACE,IAAA,KAAK,EAAEA,OAAO,GAAG,OAAH,GAAa,OAD7B;AAEE,IAAA,UAAU,EAAEC,UAFd;AAGE,IAAA,SAAS,EAAE;AAHb,KAIGF,OAJH,CADF,CADF,CAPJ,CADF;AAqBD,CAzBM","sourcesContent":["import React, { useState } from 'react'\nimport { InfoText } from '../Typography'\nimport { ToolTipProps } from './ToolTip.interface'\nimport { StyledToolTip, StyledToolTipContent, StyledToolTipWrapper } from './ToolTip.styled'\n\nexport const ToolTip: React.FC<ToolTipProps> = props => {\n  const { children, content, primary = false, limitLines = 5, ...otherProps } = props\n  const [tooltipVisible, setState] = useState(false)\n\n  return (\n    <StyledToolTipWrapper\n      onMouseEnter={() => setState(true)}\n      onMouseLeave={() => setState(false)}\n      onClick={() => setState(!tooltipVisible)}\n      {...otherProps}>\n      {children}\n      {tooltipVisible && content && (\n        <StyledToolTip primary={primary}>\n          <StyledToolTipContent primary={primary}>\n            <InfoText\n              color={primary ? 'white' : 'grey4'}\n              limitLines={limitLines}\n              textAlign={'center'}>\n              {content}\n            </InfoText>\n          </StyledToolTipContent>\n        </StyledToolTip>\n      )}\n    </StyledToolTipWrapper>\n  )\n}\n"],"file":"ToolTip.js"}