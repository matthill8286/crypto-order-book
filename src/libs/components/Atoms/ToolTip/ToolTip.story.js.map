{"version":3,"sources":["../../../../src/components/Atoms/ToolTip/ToolTip.story.tsx"],"names":["boolean","text","storiesOf","React","Link","StorybookWrapper","ToolTip","Icon","StyleguideInfoOutlined","knobs","primary","content","module","add"],"mappings":"AAAA,SAASA,OAAT,EAAkBC,IAAlB,QAA8B,wBAA9B;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,IAAT;AACA,SAASC,gBAAT;AACA,SAASC,OAAT;AAEA,SAASC,IAAT;AACA,SAASC,sBAAT,QAAuC,mCAAvC;;AAEA,IAAMC,KAAK,GAAG,SAARA,KAAQ,CAACC,OAAD,EAAqC;AACjD,SAAO;AACLC,IAAAA,OAAO,EAAEV,IAAI,CAAC,SAAD,EAAY,aAAZ,CADR;AAELS,IAAAA,OAAO,EAAEV,OAAO,CAAC,SAAD,EAAY,CAAC,CAACU,OAAd;AAFX,GAAP;AAID,CALD;;AAOAR,SAAS,CAAC,6BAAD,EAAgCU,MAAhC,CAAT,CACGC,GADH,CACO,SADP,EACkB;AAAA,SACd,oBAAC,gBAAD,QACE,oBAAC,OAAD,EAAaJ,KAAK,EAAlB,EACE,oBAAC,IAAD;AAAM,IAAA,SAAS,EAAE;AAAjB,YADF,CADF,CADc;AAAA,CADlB,EAQGI,GARH,CAQO,SARP,EAQkB;AAAA,SACd,oBAAC,gBAAD,QACE,oBAAC,OAAD,EAAaJ,KAAK,CAAC,IAAD,CAAlB,EACE,oBAAC,IAAD;AAAM,IAAA,SAAS,EAAE;AAAjB,YADF,CADF,CADc;AAAA,CARlB,EAeGI,GAfH,CAeO,MAfP,EAee;AAAA,SACX,oBAAC,gBAAD,QACE,oBAAC,OAAD,EAAaJ,KAAK,EAAlB,EACE,oBAAC,IAAD,QACE,oBAAC,sBAAD;AAAwB,IAAA,MAAM,EAAC,IAA/B;AAAoC,IAAA,KAAK,EAAC;AAA1C,IADF,CADF,CADF,CADW;AAAA,CAff","sourcesContent":["import { boolean, text } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { Link } from '@/components/Atoms/Link'\nimport { StorybookWrapper } from '@/utils/StorybookWrapper'\nimport { ToolTip } from './ToolTip'\nimport { ToolTipProps } from './ToolTip.interface'\nimport { Icon } from '@/components/Atoms/Icon'\nimport { StyleguideInfoOutlined } from '@matthill8286/atomic-icon-library'\n\nconst knobs = (primary?: boolean): ToolTipProps => {\n  return {\n    content: text('Content', 'Lorem Ipsum'),\n    primary: boolean('Primary', !!primary),\n  }\n}\n\nstoriesOf('Design System/Atoms/ToolTip', module)\n  .add('Default', () => (\n    <StorybookWrapper>\n      <ToolTip {...knobs()}>\n        <Link underline={false}>Link</Link>\n      </ToolTip>\n    </StorybookWrapper>\n  ))\n  .add('Primary', () => (\n    <StorybookWrapper>\n      <ToolTip {...knobs(true)}>\n        <Link underline={false}>Link</Link>\n      </ToolTip>\n    </StorybookWrapper>\n  ))\n  .add('Icon', () => (\n    <StorybookWrapper>\n      <ToolTip {...knobs()}>\n        <Icon>\n          <StyleguideInfoOutlined height=\"xs\" width=\"xs\" />\n        </Icon>\n      </ToolTip>\n    </StorybookWrapper>\n  ))\n"],"file":"ToolTip.story.js"}