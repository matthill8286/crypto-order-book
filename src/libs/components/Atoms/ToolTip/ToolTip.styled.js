import { styled } from "../../../styles/styled";
export var StyledToolTipWrapper = styled.div.withConfig({
  displayName: "ToolTipstyled__StyledToolTipWrapper",
  componentId: "sc-1umk7nb-0"
})(["position:relative;display:inline-block;cursor:pointer;"]);
export var StyledToolTipContent = styled.div.withConfig({
  displayName: "ToolTipstyled__StyledToolTipContent",
  componentId: "sc-1umk7nb-1"
})(["background:", ";padding:10px 0;border-radius:", ";"], function (_ref) {
  var theme = _ref.theme,
      primary = _ref.primary;
  return primary ? theme.color.grey4 : theme.color.white;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.dimension.borderRadius2;
});
export var StyledToolTip = styled.div.withConfig({
  displayName: "ToolTipstyled__StyledToolTip",
  componentId: "sc-1umk7nb-2"
})(["border-radius:", ";box-sizing:border-box;display:inline-block;position:absolute;margin-top:20px;padding:0 ", ";z-index:10;left:50%;width:120px;transform:translateX(-50%);background:", ";box-shadow:", ";&::before{content:'';box-shadow:", ";background:", ";display:block;position:absolute;width:14px;top:-7px;z-index:-1;left:50%;margin-left:-7px;height:14px;transform:rotate(45deg);}"], function (_ref3) {
  var theme = _ref3.theme;
  return theme.dimension.borderRadius2;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.spacing.base.xs;
}, function (_ref5) {
  var theme = _ref5.theme,
      primary = _ref5.primary;
  return primary ? theme.color.grey4 : theme.color.white;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.dimension.elevationLevel2;
}, function (_ref7) {
  var theme = _ref7.theme;
  return theme.dimension.elevationLevel2;
}, function (_ref8) {
  var theme = _ref8.theme,
      primary = _ref8.primary;
  return primary ? theme.color.grey4 : theme.color.white;
});
//# sourceMappingURL=ToolTip.styled.js.map