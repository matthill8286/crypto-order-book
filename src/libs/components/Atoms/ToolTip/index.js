export * from "./ToolTip";
//# sourceMappingURL=index.js.map