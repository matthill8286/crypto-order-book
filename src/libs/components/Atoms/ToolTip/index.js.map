{"version":3,"sources":["../../../../src/components/Atoms/ToolTip/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './ToolTip'\n"],"file":"index.js"}