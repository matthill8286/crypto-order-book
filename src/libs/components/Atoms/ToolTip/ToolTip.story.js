import { boolean, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Link } from "../Link";
import { StorybookWrapper } from "../../../utils/StorybookWrapper";
import { ToolTip } from "./ToolTip";
import { Icon } from "../Icon";
import { StyleguideInfoOutlined } from '@matthill8286/atomic-icon-library';

var knobs = function knobs(primary) {
  return {
    content: text('Content', 'Lorem Ipsum'),
    primary: boolean('Primary', !!primary)
  };
};

storiesOf('Design System/Atoms/ToolTip', module).add('Default', function () {
  return React.createElement(StorybookWrapper, null, React.createElement(ToolTip, knobs(), React.createElement(Link, {
    underline: false
  }, "Link")));
}).add('Primary', function () {
  return React.createElement(StorybookWrapper, null, React.createElement(ToolTip, knobs(true), React.createElement(Link, {
    underline: false
  }, "Link")));
}).add('Icon', function () {
  return React.createElement(StorybookWrapper, null, React.createElement(ToolTip, knobs(), React.createElement(Icon, null, React.createElement(StyleguideInfoOutlined, {
    height: "xs",
    width: "xs"
  }))));
});
//# sourceMappingURL=ToolTip.story.js.map