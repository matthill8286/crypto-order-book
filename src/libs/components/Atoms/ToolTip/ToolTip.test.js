import * as React from 'react';
import { mountWithTheme, renderWithThemeAndRouter } from "../../../testRenderer";
import { ToolTip } from "./ToolTip";
import { StyledToolTipContent, StyledToolTipWrapper } from "./ToolTip.styled";
describe('ToolTip', function () {
  it('renders basic', function () {
    var tree = renderWithThemeAndRouter(React.createElement(ToolTip, null));
    expect(tree).toMatchSnapshot();
  });
  it('renders default', function () {
    var comp = mountWithTheme(React.createElement(ToolTip, {
      content: "Lorem ipsum"
    }, React.createElement("span", null, "Link")));
    expect(comp.find(StyledToolTipWrapper)).toHaveLength(1);
    expect(comp).toMatchSnapshot();
  });
  it('renders primary', function () {
    var comp = mountWithTheme(React.createElement(ToolTip, {
      content: "Lorem ipsum",
      primary: true
    }, React.createElement("span", null, "Link")));
    expect(comp.find(StyledToolTipWrapper)).toHaveLength(1);
    expect(comp).toMatchSnapshot();
  });
  it('renders visible on hover and hides on leave', function () {
    var comp = mountWithTheme(React.createElement(ToolTip, {
      content: "Lorem ipsum"
    }, React.createElement("span", null, "Link")));
    expect(comp.find(StyledToolTipContent)).toHaveLength(0);
    comp.find(StyledToolTipWrapper).first().simulate('mouseenter');
    expect(comp.find(StyledToolTipContent)).toHaveLength(1);
    comp.find(StyledToolTipWrapper).first().simulate('mouseleave');
    expect(comp.find(StyledToolTipContent)).toHaveLength(0);
  });
});
//# sourceMappingURL=ToolTip.test.js.map