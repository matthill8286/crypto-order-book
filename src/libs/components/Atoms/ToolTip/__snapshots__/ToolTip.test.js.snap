// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolTip renders basic 1`] = `
<div
  className="ToolTipstyled__StyledToolTipWrapper-sc-1umk7nb-0 hrfefP"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
